### DISCUSSION TITLE
dtail Usage Issues

### DISCUSSION THREAD

**Discussion by EnsonoStevenPerva** [IBM/zoau-samples#24] [Q&A] [answered]
Using `dtail` with ZOAU version 2021/08/05 22:12:58 CUT V1.2.0, I am experiencing strange behavior with the command. I am creating a member as follows:

`decho 'Hello' '<My_ID>.ZOAU.PDSE(MSG)'`

And then to confirm that the message is there, I am trying the following:

`dtail "<My_ID>.ZOAU.PDSE(MSG)"`

I am getting no response to this command. I believe this behavior is unexpected because `dtail` is supposed to default to the last ten lines. If I provide a member with at least 10 lines:

```
decho "Line 1
Line 2
Line 3
Line 4
Line 5
Line 6
Line 7
Line 8
Line 9
Line A" "<My_ID>.ZOAU.PDSE(MSG2)"
```

`dtail` provides the expected response of showing the last ten lines. An observation is that `dtail` will only provide results if the target has at least the requested number of lines showing:

`dtail -n -1 "<My_ID>.ZOAU.PDSE(MSG)"`

I would expect that in the event there are less-than the number of requested lines, that `dtail` will show the entire result set instead of nothing.

**Comment by MikeFultonDev** [comment, marked as answer]
I did this:
```
dtouch ibmuser.pds
IBMUSER:/app/temp/ibmuser: >decho "Hello world" ibmuser.pds\(out\)
IBMUSER:/app/temp/ibmuser: >dtail ibmuser.pds\(out\)
Hello world                                                                     
IBMUSER:/app/temp/ibmuser: >zoaversion
2021/03/26 15:44:32 CUT V1.1.1
```

so it seems to work ok for me with 1.1.1
I'll try it with 1.2.0 as well unless someone already has done that.

**Reply by EnsonoStevenPerva** [reply]
Odd, that just worked for me on 1.2.0. So I tried my previous example:

```
IBMUSER:/u/IBMUSER: >dtouch IBMUSER.pds
IBMUSER:/u/IBMUSER: >decho "Hello world" IBMUSER.pds\(out\)
IBMUSER:/u/IBMUSER: >dtail IBMUSER.pds\(out\)
Hello world
IBMUSER:/u/IBMUSER: >zoaversion
2021/08/05 22:12:58 CUT V1.2.0
IBMUSER:/u/IBMUSER: >decho 'Hello' 'IBMUSER.ZOAU.PDSE(MSG)'
IBMUSER:/u/IBMUSER: >dtail "IBMUSER.ZOAU.PDSE(MSG)"
IBMUSER:/u/IBMUSER: >
```

I wonder if the quoting is doing something odd...

**Reply by MikeFultonDev** [reply]
Hm. Looks ok to me. You can do `cat “//‘IBM user.zoau.pdse(msg)’”` to make sure it isn’t decho that is failing.

**Reply by EnsonoStevenPerva** [reply]
Here's what I've got:
```
IBMUSER:/u/IBMUSER: >cat "//'IBMUSER.ZOAU.PDSE(MSG)'"
Hello
IBMUSER:/u/IBMUSER: >dtail IBMUSER.ZOAU.PDSE\(MSG\)
IBMUSER:/u/IBMUSER: >dtail -n -1 IBMUSER.ZOAU.PDSE\(MSG\)
Hello
IBMUSER:/u/IBMUSER: >
```

**Reply by MikeFultonDev** [reply]
I can't reproduce this @EnsonoStevenPerva . What are the dataset attributes on your PDSE and what are the characters in your PDS member to the right of 'Hello' if you go into ISPF edit and do a 'hex on' ?

**Reply by EnsonoStevenPerva** [reply]
I can no longer reproduce this in both of my environments. My original dataset still has this issue, but any attempt at reproducing it is unsuccessful. I am going to chalk this up to some error I made. It seems the PDSE I created earlier has very different attributes than any PDSE I am creating now.

My original dataset is a VB PDS with 1028 LRECL and 6144 block size.
Any new dataset is a FB PDSE with 80 LRECL and 32720 block size.

I suspect I did something erroneously for the original dataset with `dtouch`. I thought I took the defaults, but this discovery would lead me to believe I didn't.

Sorry for the goose chase here.

**Comment by ddimatos** [comment]
Hey all, I was not able to recreate either, I used zoau 1.2.0 though with quotes and no quotes. 

```
ZOAUSER:/u/zoauser #>dtouch ibmuser.pds
ZOAUSER:/u/zoauser #>decho "Hello world" ibmuser.pds\(out\)
ZOAUSER:/u/zoauser #>dtail ibmuser.pds\(out\)
Hello world
ZOAUSER:/u/zoauser #>dtail -n -1 ibmuser.pds\(out\)
Hello world
```

```
ZOAUSER:/u/zoauser #>dtouch "user.pds"
ZOAUSER:/u/zoauser #>decho "Hello world" "user.pds(out)"
ZOAUSER:/u/zoauser #>dtail "imbuser.pds(out)"
Hello world
ZOAUSER:/u/zoauser #>dtail -n -1 "imbuser.pds(out)"
Hello world
ZOAUSER:/u/zoauser #>zoaversion
```

```
2021/08/05 22:12:58 CUT V1.2.0
```
